libr/anal/dwarf_process.c

```diff
@@ -0,0 +1,1376 @@
+#include "base_types.h"
+#include <sdb.h>
+#include <r_anal.h>
+#include <r_bin_dwarf.h>
+#include <string.h>
+
+typedef struct dwarf_parse_context_t {
+	const RAnal *anal;
+	const RBinDwarfDie *all_dies;
+	const ut64 count;
+	Sdb *sdb;
+	HtUP *die_map;
+	char *lang; // for demangling
+} DwContext;
+
+typedef struct dwarf_function_t {
+	ut64 addr;
+	const char *name;
+	const char *signature;
+	bool is_external;
+	bool is_method;
+	bool is_virtual;
+	bool is_trampoline; // intermediary in making call to another func
+	ut8 access; // public = 1, protected = 2, private = 3, if not set assume private
+	ut64 vtable_addr; // location description
+	ut64 call_conv; // normal || program || nocall
+} DwFunction;
+
+typedef enum dwarf_location_kind {
+	LOCATION_UNKNOWN = 0,
+	LOCATION_GLOBAL = 1,
+	LOCATION_BASE = 2, /* base ptr */
+	LOCATION_STACK = 3, /* stack ptr */
+	LOCATION_REGISTER = 4,
+} DwfVariableLocationKind;
+typedef struct dwarf_var_location_t {
+	DwfVariableLocationKind kind;
+	st64 offset;
+	ut64 address;
+} DwfVariableLocation;
+
+typedef struct dwarf_variable_t {
+	DwfVariableLocation *location;
+	const char *name;
+	char *type;
+} DwfVariable;
+
+
+
+static inline bool is_type_tag(ut64 tag_code) {
+	return (tag_code == DW_TAG_structure_type ||
+		tag_code == DW_TAG_enumeration_type ||
+		tag_code == DW_TAG_class_type ||
+		tag_code == DW_TAG_subprogram ||
+		tag_code == DW_TAG_union_type ||
+		tag_code == DW_TAG_base_type ||
+		tag_code == DW_TAG_typedef);
+}
+
+/**
+ * @brief Finds index of a particular attribute of a DIE
+ * 
+ * @param die 
+ * @param attr_name 
+ * @return st32 Index, -1 if nothing found
+ */
+static st32 find_attr_idx(const RBinDwarfDie *die, st32 attr_name) {
+	st32 i;
+	r_return_val_if_fail (die, -1);
+	for (i = 0; i < die->count; i++) {
+		if (die->attr_values[i].attr_name == attr_name) {
+			return i;
+		}
+	}
+	return -1;
+}
+
+/**
+ * @brief Prepends string before a last occurence of character c
+ * 	      Used to replicate proper C declaration for function pointers
+ * @param sb 
+ * @param s 
```

Comment:
Is this suposed to help on understanding the purpose of the parameters? It doesnt seems necessary or useful to me